### ISSUE TITLE
HGVS del - setting to display or not, and hide in SA Path

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#498]
Karin says: The HGVS nomenclature recommends not showing the deleted bases, e.g. e.g. NC_000007.14:g.117603750del as opposed to NC_000007.14:g.117603750delC

We've gone back and forth about this so much

* Make a setting to decide whether to show deleted bases or not
* Configure this to be disabled in SA Path

**Comment by davmlaw** [comment]
Renamed max_allele_length => max_ref_length (as it only ever changed ref not alt)
Renamed VARIANT_CLASSIFICATION_MAX_FULL_ALLELE_LENGTH => VARIANT_CLASSIFICATION_MAX_REFERENCE_LENGTH


Renamed HGVS_REQUIRE_REFERENCE_BASES => CLINGEN_ALLELE_REGISTRY_REQUIRE_REF_ALLELE - as only used by ClinGen

Instead of hardcoding 10 (for max ref length) we now have a setting:

HGVS_MAX_REF_ALLELE_LENGTH = 10 # Will be 0 for SA Path


HGVSExtra:

    * Fixed bug where can_shrink_long_ref never passed length but relied on defaults
    * Moved can_shrink_long_ref / format_hgvs_remove_long_ref from HGVSMatcher into HGVSExtra (was only ever called from there)
    
 HF to SAP will have to be manually done - as this file has massively diverged.

Change is https://github.com/SACGF/variantgrid/commit/9b951ca0da01ddac1ed300f7fd62a7d475dfc2df

Basically just need the setting, and modify HGVSExtra.format()

**Comment by TheMadBug** [comment]
So the main driver behind showing optional bases in Shariant is for Alissa import which currently requires them.
The code that compares c.hgvs to see if our resolving needs more investigation does a specific check for such things

hgvs.py
```
DIFF_RAW_CGVS_EXPANDED
# 'The del|ins|dup is explicit'
```
but this is specifically marked as minor and will not cause warnings.

As long as cached_chgvs_37_full, cached_chgvs_38_full still always have optional bases, then this should be fine.

**Comment by davmlaw** [comment]
Yep if you leave ```HGVS_MAX_REF_ALLELE_LENGTH = 10 ``` there should be no change in behavior - only that SAP want to be able to zero it out

**Comment by davmlaw** [comment]
Different settings on variantgrid.com and SAP

**Comment by davmlaw** [comment]
This originally failed in SAP - NC_000007.14:g.117603750del - still shows C for g.HGVS - due to a few ```max_ref_length=10``` left in HGVS code.

Patched in 0e7f1710c561734cc6ff7a51c197b9f0295a8043 - now tests ok

**Comment by davmlaw** [comment]
KK called said testing passed 17/12/2021
Dave verified in prod, variant details page and classification has no base for dels,
